component/src/main/java/org/wso2/extension/siddhi/io/file/FileSource.java

```diff
@@ -0,0 +1,539 @@
+    }
+
+    private void createInitialSourceConf() {
+        fileSourceConfiguration.setDirURI(dirUri);
+        fileSourceConfiguration.setFileURI(fileUri);
+        fileSourceConfiguration.setMoveAfterProcessUri(moveAfterProcess);
+        fileSourceConfiguration.setBeginRegex(beginRegex);
+        fileSourceConfiguration.setEndRegex(endRegex);
+        fileSourceConfiguration.setMode(mode);
+        fileSourceConfiguration.setActionAfterProcess(actionAfterProcess);
+        fileSourceConfiguration.setTailingEnabled(Boolean.parseBoolean(tailing));
+        fileSourceConfiguration.setFilePollingInterval(filePollingInterval);
+        fileSourceConfiguration.setDirPollingInterval(dirPollingInterval);
+        fileSourceConfiguration.setActionAfterFailure(actionAfterFailure);
+        fileSourceConfiguration.setMoveAfterFailure(moveAfterFailure);
+        fileSourceConfiguration.setRequiredProperties(requiredProperties);
+    }
+
+    private void updateSourceConf() {
+        fileSourceConfiguration.setFilePointer(filePointer);
+        fileSourceConfiguration.setTailedFileURI(tailedFileURI);
+    }
+
+    private Map<String, String> getFileSystemServerProperties() {
+        Map<String, String> map = new HashMap<>();
+
+        map.put(Constants.TRANSPORT_FILE_DIR_URI, dirUri);
+        if (actionAfterProcess != null) {
+            map.put(Constants.ACTION_AFTER_PROCESS_KEY, actionAfterProcess.toUpperCase(Locale.ENGLISH));
+        }
+        map.put(Constants.MOVE_AFTER_PROCESS_KEY, moveAfterProcess);
+        map.put(Constants.POLLING_INTERVAL, dirPollingInterval);
+        map.put(Constants.FILE_SORT_ATTRIBUTE, Constants.NAME);
+        map.put(Constants.FILE_SORT_ASCENDING, Constants.TRUE);
+        map.put(Constants.CREATE_MOVE_DIR, Constants.TRUE);
+        map.put(Constants.ACK_TIME_OUT, "5000");
+
+        if (Constants.BINARY_FULL.equalsIgnoreCase(mode) ||
+                Constants.TEXT_FULL.equalsIgnoreCase(mode)) {
+            map.put(Constants.READ_FILE_FROM_BEGINNING, Constants.TRUE);
+        } else {
+            map.put(Constants.READ_FILE_FROM_BEGINNING, Constants.FALSE);
+        }
+        if (actionAfterFailure != null) {
+            map.put(Constants.ACTION_AFTER_FAILURE_KEY, actionAfterFailure);
+        }
+        if (moveAfterFailure != null) {
+            map.put(Constants.MOVE_AFTER_FAILURE_KEY, moveAfterFailure);
+        }
+        return map;
+    }
+
+    private void validateParameters() {
+        if (Constants.TEXT_FULL.equalsIgnoreCase(mode) || Constants.BINARY_FULL.equalsIgnoreCase(mode)) {
+            if (isTailingEnabled) {
+                throw new SiddhiAppRuntimeException("Tailing can't be enabled in '" + mode + "' mode.");
+            }
+        }
+
+        if (isTailingEnabled && moveAfterProcess != null) {
+            throw new SiddhiAppRuntimeException("'moveAfterProcess' cannot be used when tailing is enabled. " +
+                    "Hence stopping the SiddhiApp. ");
+        }
```

Comment:
Log siddhi app name also so that the user knows from which siddhi app this error is coming from.

Reply:
Done by 0b41464d4ebc6683c78635fe198ceba602f845e6